### ISSUE TITLE
Mtimproxy crashed while deleting flow

### ISSUE THREAD

**Issue by yasoob** [mitmproxy/mitmproxy#437]
I was testing mitmproxy. I hit the ```d``` key and it crashed:

```
yasoob@yasoob-laptop:~$ mitmproxy -U http://127.0.0.1:8118
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/libmproxy/console/__init__.py", line 610, in run
    self.ui.run_wrapper(self.loop)
  File "/usr/local/lib/python2.7/dist-packages/urwid/display_common.py", line 757, in run_wrapper
    return fn()
  File "/usr/local/lib/python2.7/dist-packages/libmproxy/console/__init__.py", line 841, in loop
    k = self.view.keypress(size, k)
  File "/usr/local/lib/python2.7/dist-packages/urwid/container.py", line 1128, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/usr/local/lib/python2.7/dist-packages/libmproxy/console/flowlist.py", line 270, in keypress
    return urwid.ListBox.keypress(self, size, key)
  File "/usr/local/lib/python2.7/dist-packages/urwid/listbox.py", line 987, in keypress
    key = focus_widget.keypress((maxcol,),key)
  File "/usr/local/lib/python2.7/dist-packages/libmproxy/console/flowlist.py", line 147, in keypress
    self.state.delete_flow(self.flow)
  File "/usr/local/lib/python2.7/dist-packages/libmproxy/console/__init__.py", line 331, in delete_flow
    ret = flow.State.delete_flow(self, f)
  File "/usr/local/lib/python2.7/dist-packages/libmproxy/flow.py", line 434, in delete_flow
    self._flow_list.remove(f)
ValueError: list.remove(x): x not in list

mitmproxy has crashed!
Please lodge a bug report at: https://github.com/mitmproxy/mitmproxy
Shutting down...
```

Background
----------------

I was trying to clear the flow list by hitting ```C```. Some flows were cleared but the others remained there. I typed ```?``` and looked at the flow deleting key. I typed ```q``` to return back to main screen and hit ```d``` after going to the last flow on my screen. After that, BOOM! It crashed! 

If you need any more information just let me know. :+1:

**Comment by mhils** [comment]
Hi there,

thanks for the report. Which version of mitmproxy do you use? 0.11.2 or 0.11.3? 
I wasn't able to reproduce this - can you trigger this reliably? In particular, "Some flows were cleared but the others remained there." - if I press `C`, no flows remain for me.

**Comment by mhils** [comment]
Íf anyone can reproduce this issue, please let us know.

**Comment by rtoma** [comment]
Hit same bug on yosemite with mitmproxy 0.11.1 from brew:
```
Traceback (most recent call last):
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/lib/python2.7/site-packages/libmproxy/console/__init__.py", line 610, in run
    self.ui.run_wrapper(self.loop)
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/vendor/lib/python2.7/site-packages/urwid/raw_display.py", line 268, in run_wrapper
    return fn()
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/lib/python2.7/site-packages/libmproxy/console/__init__.py", line 841, in loop
    k = self.view.keypress(size, k)
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/vendor/lib/python2.7/site-packages/urwid/container.py", line 1124, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/lib/python2.7/site-packages/libmproxy/console/flowlist.py", line 270, in keypress
    return urwid.ListBox.keypress(self, size, key)
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/vendor/lib/python2.7/site-packages/urwid/listbox.py", line 987, in keypress
    key = focus_widget.keypress((maxcol,),key)
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/lib/python2.7/site-packages/libmproxy/console/flowlist.py", line 147, in keypress
    self.state.delete_flow(self.flow)
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/lib/python2.7/site-packages/libmproxy/console/__init__.py", line 331, in delete_flow
    ret = flow.State.delete_flow(self, f)
  File "/usr/local/Cellar/mitmproxy/0.11.1/libexec/lib/python2.7/site-packages/libmproxy/flow.py", line 434, in delete_flow
    self._flow_list.remove(f)
ValueError: list.remove(x): x not in list

mitmproxy has crashed!
Please lodge a bug report at: https://github.com/mitmproxy/mitmproxy
Shutting down...
```

**Comment by mhils** [comment]
This issue should be fixed in 0.11.3. Could you please re-check with the latest version?

Cheers,
Max

**Comment by rtoma** [comment]
Would love to, but brew is still on 0.11.1. Once .3 is available (on brew) I will check and get back to you.

Btw, props on your tool! Its excellent for reverse engineering icloud protocol.
